hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestingUtility.java

```diff
@@ -1049,6 +1049,21 @@ public MiniHBaseCluster startMiniCluster(int numSlaves) throws Exception {
     return startMiniCluster(option);
   }
 
+  /**
+   * Start up a minicluster of hbase, dfs and zookeeper clusters with given slave node number and
+   * a set of master ports. All other options will use default values, defined in
+   * {@link StartMiniClusterOption.Builder}.
+   * @param numSlaves slave node number, for both HBase region server and HDFS data node.
+   * @see #startMiniCluster(StartMiniClusterOption option)
+   * @see #shutdownMiniDFSCluster()
+   */
+  public MiniHBaseCluster startMiniClusterWithMasterPorts(int numSlaves) throws Exception {
+    StartMiniClusterOption option = StartMiniClusterOption.builder()
+        .numRegionServers(numSlaves).numDataNodes(numSlaves)
+        .masterPorts(randomFreePorts(1)).build();
```

Comment:
In the test environment, why is it that we don't use random ports always?

Reply:
Re-did the patch to be much simpler. Not valid anymore.